### PR TITLE
[soft-assert] Add ability to set description for known issue

### LINKED ISSUE #2011: [known-issues] Add possibility to specify comment alongside with the known issue
**Is your feature request related to a problem? Please describe.**
Currently, there is no possibility to put any information like a comment or additional details for the known issue. We. could communicate using the Jira ticket that is for sure, but it doesn't cover the case when you need to put some info along the known issue, so it may be seen in the report

**Describe the solution you'd like**
Add optional "comment"

```json

"VVD-1": {
    "type": "Internal",
    "assertionPattern": ".*Doctor Who.*",
    "scenarioPattern": "Verify step Then I verify assertions matching '.+'",
    "storyPattern": "AssertStepsTests",
    "comment": "With the help of the known we issue we verify functionality x"
  }

```
And allure
```
1) Known issue: VVD-1 (Type: INTERNAL. Status: UNKNOWN, Comment: With the help of the known we issue we verify functionality x). Number of elements found by ' Accessibility Id: 'Login'; Visibility: VISIBLE;' is equal to 1 [Expected: number of elements is a value equal to <1> Actual: <0>]
```

**Describe alternatives you've considered**
We could use the identifier as a workaround but it looks ugly

### PR DESCRIPTION
**Pull request by uarlouski**
Closes #2011

### CODE REVIEW

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 1]
```diff
@@ -117,7 +117,8 @@ The file represents a collection of JSON elements describing known issues.
             "var":  "value-.*"
         },
         "failScenarioFast": true, <8>
-        "failStoryFast": true <9>
+        "failStoryFast": true, <9>
+        "description": "Missing listener on a button" <10>
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:121]: it's better to have example which matches the issue, in general issue describes some REST API issue:
```suggestion
        "description": "Missing endpoint configuration on server side" <10>
```

File: docs/modules/ROOT/pages/tests-configuration.adoc [hunk 2]
```diff
@@ -130,6 +131,7 @@ The file represents a collection of JSON elements describing known issues.
 <7> Variable patterns. Patterns to match variables.
 <8> Skip the rest of scenario in case of failure identified as this known issue
 <9> Skip the rest of story in case of failure identified as this known issue
+<10> Known issue description
```
Comment (valfirst) [inline docs/modules/ROOT/pages/tests-configuration.adoc:134]:
```suggestion
<10> Known issue description, it's used to give the meaningful explanation of the issue if it's not tracked in bug tracking system or if the issue is too complicated and to improve overall transparency
```

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java [hunk 3]
```diff
@@ -30,6 +30,7 @@
     private Pattern assertionCompiledPattern;
     private boolean failTestCaseFast;
     private boolean failTestSuiteFast;
+    private String description;
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java:33]:
```suggestion
    private Optional<String> description;
```
?

File: vividus-tests/src/main/resources/known-issues.json [hunk 4]
```diff
@@ -62,13 +62,15 @@
     "failScenarioFast": true
   },
   "VVD-10": {
+    "description": "Catch debug step failure",
```
Comment (valfirst) [inline vividus-tests/src/main/resources/known-issues.json:65]:
```suggestion
    "description": "Hide debug step failure",
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java (hunk 3 above)
Comment (uarlouski) [reply vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java:33]: do you suggest to convert optional in setter? I'm not sure if is supports auto setting of non-specified in JSON fields to Optional type, i'll check

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java [hunk 5]
```diff
@@ -84,6 +85,26 @@ public void setFailTestSuiteFast(boolean failTestSuiteFast)
...
+    {
+        return description;
+    }
+
+    public void setDescription(String description)
+    {
+        this.description = description;
+    }
+
+    public void setAssertionCompiledPattern(Pattern assertionCompiledPattern)
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java:98]: where is it used and why is it needed?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java [hunk 6]
```diff
@@ -84,6 +85,26 @@ public void setFailTestSuiteFast(boolean failTestSuiteFast)
...
+    {
+        this.assertionCompiledPattern = assertionCompiledPattern;
+    }
+
+    public void setAdditionalCompiledPatterns(Map<String, Pattern> additionalCompiledPatterns)
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java:103]: where is it used and why is it needed?

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java (hunk 5 above)
Comment (uarlouski) [reply vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java:98]: missclick in code generation :)

File: vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java (hunk 6 above)
Comment (uarlouski) [reply vividus-soft-assert/src/main/java/org/vividus/softassert/issue/KnownIssueIdentifier.java]: missclick in code generation :)

**Review by valfirst — APPROVED** [review]

**Pull request merged**